File: E-commerce/src/main/java/com/info/ecommerce/modules/payment/entity/PaymentCallbackLog.java
package com.info.ecommerce.modules.payment.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 支付回調記錄實體
 * 記錄所有支付閘道的回調請求，用於除錯和追蹤
 */
@Entity
@Table(name = "payment_callback_log")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PaymentCallbackLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    /**
     * 支付閘道類型
     */
    @Column(name = "gateway", nullable = false, length = 50)
    private String gateway;

    /**
     * 訂單編號
     */
    @Column(name = "order_number", length = 100)
    private String orderNumber;

    /**
     * 交易 ID
     */
    @Column(name = "transaction_id", length = 100)
    private String transactionId;

    /**
     * 回調狀態（SUCCESS, FAILED, ERROR）
     */
    @Column(name = "status", nullable = false, length = 50)
    private String status;

    /**
     * 原始請求參數（JSON 格式）
     */
    @Column(name = "raw_params", columnDefinition = "NVARCHAR(MAX)")
    private String rawParams;

    /**
     * 解析後的響應數據（JSON 格式）
     */
    @Column(name = "parsed_response", columnDefinition = "NVARCHAR(MAX)")
    private String parsedResponse;

    /**
     * 處理結果
     */
    @Column(name = "process_result", length = 500)
    private String processResult;

    /**
     * 錯誤訊息
     */
    @Column(name = "error_message", columnDefinition = "NVARCHAR(1000)")
    private String errorMessage;

    /**
     * 請求 IP
     */
    @Column(name = "request_ip", length = 50)
    private String requestIp;

    /**
     * User-Agent
     */
    @Column(name = "user_agent", length = 500)
    private String userAgent;

    /**
     * 處理時間（毫秒）
     */
    @Column(name = "process_time_ms")
    private Long processTimeMs;

    /**
     * 建立時間
     */
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: frontend/src/api/paymentCallbackLog.ts
/**
 * 支付回調記錄相關 API
 * @module PaymentCallbackLogAPI
 */

import axios from './axios'
import type { ApiResponse, PageResponse } from './types'

/**
 * 支付回調記錄介面
 */
export interface PaymentCallbackLog {
  /** 記錄 ID */
  id?: number
  /** 支付閘道類型 */
  gateway: string
  /** 訂單編號 */
  orderNumber?: string
  /** 交易 ID */
  transactionId?: string
  /** 回調狀態 */
  status: string
  /** 原始請求參數（JSON） */
  rawParams?: string
  /** 解析後的響應數據（JSON） */
  parsedResponse?: string
  /** 處理結果 */
  processResult?: string
  /** 錯誤訊息 */
  errorMessage?: string
  /** 請求 IP */
  requestIp?: string
  /** User-Agent */
  userAgent?: string
  /** 處理時間（毫秒） */
  processTimeMs?: number
  /** 建立時間 */
  createdAt?: string
}

/**
 * 支付回調記錄 API 服務
 */
export const paymentCallbackLogApi = {
  /**
   * 取得回調記錄列表（分頁）
   */
  getCallbackLogs: (params?: { page?: number; size?: number }) => {
    return axios.get<any, ApiResponse<PageResponse<PaymentCallbackLog>>>('/payment/callback-logs', { params })
  },

  /**
   * 根據 ID 取得回調記錄詳情
   */
  getCallbackLog: (id: number) => {
    return axios.get<any, ApiResponse<PaymentCallbackLog>>(`/payment/callback-logs/${id}`)
  },

  /**
   * 根據訂單編號查詢回調記錄
   */
  getCallbackLogsByOrderNumber: (orderNumber: string) => {
    return axios.get<any, ApiResponse<PaymentCallbackLog[]>>(`/payment/callback-logs/order/${orderNumber}`)
  },

  /**
   * 根據交易 ID 查詢回調記錄
   */
  getCallbackLogsByTransactionId: (transactionId: string) => {
    return axios.get<any, ApiResponse<PaymentCallbackLog[]>>(`/payment/callback-logs/transaction/${transactionId}`)
  },

  /**
   * 根據支付閘道類型查詢回調記錄
   */
  getCallbackLogsByGateway: (gateway: string, params?: { page?: number; size?: number }) => {
    return axios.get<any, ApiResponse<PageResponse<PaymentCallbackLog>>>(`/payment/callback-logs/gateway/${gateway}`, { params })
  }
}

export default paymentCallbackLogApi

